src/module-analyzer.js

```diff
@@ -240,17 +240,24 @@ export class ModuleAnalyzer {
         continue;
       }
 
+      // This is an unexpected behavior for inheritance as it will walk through the whole prototype chain
+      // to look for metadata. Should be `getOwn` instead. Though it's subjected to a breaking changes change
       resourceTypeMeta = metadata.get(metadata.resource, exportedValue);
```

Comment:
Similar to the other usage of `metadata.get`, this would eliminate the needs for a derived class to have their own resource metadata in inheritance scenarios if it was `getOwn`. Could be a good target for imrpovement, but it could be a breaking change if made into `getOwn`  @EisenbergEffect  @jods4

Reply:
@EisenbergEffect @jods4 after thinking a bit more about this, i think its safe to change it to `getOwn`. Old apps that have derived classes with metadata ready will continue to work. New app will work without having define its metadata upfront. But old apps that rely on it to break here (not defining any metadata in derived class but has some in base class) will break or magically work in some unexpected way. Not sure anyone would have that though